### PR TITLE
Dialog for moving event participants

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR creates a dialog for managing participants of multiple events side by side.

## Screenshots
image

## Changes
* Adds `EventParticipantsModal` to generate a list of pending `ParticipantOp` objects, that can then be passed to the new `moveParticipants` RPC for execution against the Zetkin API.
* Modifies the `SelectionBar` to allow opening the modal from there
* Fixes a bug in the `mockState()` function, caused by the fact that it wasn't pure (it was using the same module-scope object over and over again, so if a test modified it that change would carry over to the next test)

## Notes to reviewer
There is a bug which causes events in the multi-event popper not to update after making changes with this dialog. But I have verified that the problem is not with in this new feature (the store updates correctly) but in the way the popper keeps it's state. Will file separate bug once this is merged!

## Related issues
Undocumented

### CODE REVIEW

File: src/features/events/hooks/useParticipantPool.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      const { result, rerender } = renderHook(() => useParticipantPool(), {
+        wrapper: makeWrapper(store),
+      });
+
+      act(() => {
+        result.current.moveFrom(1, 11);
+        result.current.moveTo(1, 11);
+      });
+
+      rerender();
```
Comment (ziggabyte) [inline src/features/events/hooks/useParticipantPool.spec.ts:70]: have not seen this pattern in our tests before! cool!

File: src/features/events/rpc/moveParticipants.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        await apiClient.put(
+          `/api/orgs/${orgId}/actions/${eventId}/participants/${personId}`
+        );
+      } else {
+        await apiClient.delete(
+          `/api/orgs/${orgId}/actions/${eventId}/participants/${personId}`
+        );
+      }
+    } catch (err) {
+      // Just ignore
```
Comment (ziggabyte) [inline src/features/events/rpc/moveParticipants.ts:43]: 1z2tmc

File: src/features/events/components/EventParticipantsModal/EventsListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  if (status === EventState.OPEN) {
+    color = STATUS_COLORS.GREEN;
+  } else if (status === EventState.ENDED) {
+    color = STATUS_COLORS.RED;
+  } else if (status === EventState.SCHEDULED) {
+    color = STATUS_COLORS.BLUE;
+  } else if (status === EventState.CANCELLED) {
+    color = STATUS_COLORS.ORANGE;
+  }
+  const classes = useStyles({ color, selected });
```
Comment (ziggabyte) [inline src/features/events/components/EventParticipantsModal/EventsListItem.tsx:86]: i am so looking forward to making this pattern better

File: src/features/events/components/EventParticipantsModal/EventsListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      onClick={() => {
+        onSelect();
+      }}
+    >
+      <Box className={classes.left}>
+        <Tooltip title={getStatusDotLabel({ color })}>
+          <Box className={classes.dot} />
+        </Tooltip>
+        <Box>
+          <Typography color={theme.palette.text.primary}>{title}</Typography>
```
Comment (ziggabyte) [inline src/features/events/components/EventParticipantsModal/EventsListItem.tsx:110]: is `text.primary` not the automatic color for `Typography` text content?

File: src/features/events/components/EventParticipantsModal/ParticipantsSection.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          />
+        )}
+      </Box>
+
+      <Box mb={4}>
+        <Typography variant="h5">
+          <Msg id={messageIds.participantsModal.participants.headers.pending} />
+        </Typography>
+        {!pendingParticipants.length && (
+          <Typography variant="body1">
```
Comment (ziggabyte) [inline src/features/events/components/EventParticipantsModal/ParticipantsSection.tsx:43]: Is `body1` not the default text variant? Does adding this make a difference?

**Review by ziggabyte — APPROVED** [review]
Playing with it I think it works as expected :D 

I left a few comments, don't think they matter really just thougths i had while reading :)

File: src/features/events/components/EventParticipantsModal/EventsListItem.tsx (hunk 4 above)
Comment (richardolsson) [reply src/features/events/components/EventParticipantsModal/EventsListItem.tsx:110]: I think you're right, and it's just me feeling uncomfortable using `Typography` without any attributes. But I will remove the unnecessary props!

Comment (richardolsson) [reply src/features/events/components/EventParticipantsModal/EventsListItem.tsx]: Fixed in 1c3ef2d56946d3de389598bcf7b40068a6267011

File: src/features/events/components/EventParticipantsModal/ParticipantsSection.tsx (hunk 5 above)
Comment (richardolsson) [reply src/features/events/components/EventParticipantsModal/ParticipantsSection.tsx]: Fixed in 1c3ef2d56946d3de389598bcf7b40068a6267011

**Review by ziggabyte — APPROVED** [review]
i approve of these changes

**Pull request merged**